code/modules/hydroponics/seeds.dm

```diff
@@ -616,3 +616,24 @@
 
 /obj/item/grown/get_plant_seed()
 	return seed
+
+/obj/item/seeds/proc/perform_reagent_pollination(obj/item/seeds/donor)
+	var/list/datum/plant_gene/reagent/valid_reagents = list()
+	for(var/datum/plant_gene/reagent/donor_reagent in donor.genes)
+		var/repeated = FALSE
+		for(var/datum/plant_gene/reagent/receptor_reagent in genes)
+			if(donor_reagent.reagent_id == receptor_reagent.reagent_id)
```

Comment:
If you invert this check and add an early continue, you could also remove `var/repeated` and move `valid_reagents += donor_reagent` to the early continue as well. That'll really clean up these nested loops

Reply:
i don't think this would work because you need to check all the receptor genes, an early continue would just re add the bug i fixed of the receptor and donor sharing genes both have, what i can do is break the inner loop once repeated is set to TRUE